src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -272,7 +282,12 @@ public static DisposeMethodKind GetDisposeMethodKind(this IMethodSymbol method,
             {
                 if (IsDisposeImplementation(method, iDisposable) ||
                     (Equals(method.ContainingType, iDisposable) &&
-                     method.HasDisposeMethodSignature()))
+                     method.HasDisposeMethodSignature())
+#if NET_ANALYZERS
```

Comment:
Why `#if NET_ANALYZERS`? This will cause this fix not to appear in FxCop analyzers package. I think you probably `#if HAS_IOPERATION`, which should prevent it.

```suggestion
#if HAS_IOPERATION
```

Reply:
I was not sure what conditional to use for compatibility, and neither of the conditionals are documented.

I used this `#if` because, apparently, not all projects that include these files reference a version of `Microsoft.CodeAnalysis.Common`, that has `IsRefLikeType` property on `ITypeSymbol`.

Can you confirm, that `HAS_IOPERATION` is more meaningful for this use case, and why?

P.S. roslyn-analyzers team should probably document the conditionals and the rationale behind them somewhere it could be easily found.

Reply:
Yeah, these directives are really unfortunate, and needed primarily because we have couple of analyzer packages that target a very early version of Microsoft.CodeAnalysis and import this shared project. I can clean this up in a follow-up PR to make it more explicit. For now, switching to `HAS_IOPERATION` will ensure that such analyzer packages on older Microsoft.CodeAnalysis will be excluded.

Reply:
This suggestion is not working. The following projects seem to have this outdated dependency lacking `IsRefLikeType` and define `HAS_IOPERATION`: PerformanceSensitiveAnalyzers and BannedApiAnalyzers

Reply:
@mavasani any suggestions regarding `HAS_IOPERATION` here in `PerformanceSensitiveAnalyzers` and `BannedApiAnalyzers`?

Reply:
Those packages should not be needing these APIs. It is fine if this functionality does not exist for those analyzer packages.

Reply:
@mavasani problem is they define `HAS_IOPERATION`, so I can not use it to skip `IsRefLikeType` check there.

Reply:
Ah, thanks for the clarification. It seems reasonable to define a new repo wide variable, say `NOT_ON_LATEST_MICROSOFT_CODE_ANALYSIS` or some such, and all packages which override their referenced Microsoft.CodeAnalysis package reference version to a lower one then defined in the repo's Versions.props can define this constant in their project file.

Reply:
@mavasani ok, added `CODEANALYSIS_V3_OR_BETTER` conditional compilation symbol.